src/bin/pg_autoctl/keeper_pg_init.c

```diff
@@ -86,6 +88,198 @@ keeper_pg_init_fsm(Keeper *keeper, KeeperConfig *config)
+	char defaultPosgresqlConfPath[MAXPGPATH];
+	char defaultHbaConfPath[MAXPGPATH];
+	char defaultIdentConfPath[MAXPGPATH];
+	char identConfPath[MAXPGPATH];
+
+	get_absolute_path(config->pgSetup.pgdata, pgdata, MAXPGPATH);
+
+	/* check if postgresql.conf exists in pgdata. there is no separate check for pg_hba and pg_ident  */
+	join_path_components(posgresqlConfPath, pgdata, "postgresql.conf");
+	postgresqlConfExists = file_exists(posgresqlConfPath);
+	if (postgresqlConfExists)
+	{
+		return true;
+	}
+
+	/* check if pgdata is in default path. */
+	if (strstr(pgdata, "/var/lib/postgresql") != pgdata)
+	{
+		log_error("Cannot determine configuration file location for this instance. pgdata = %s", pgdata);
+		return false;
+	}
+
+	/*
+	 * We are trying to translate /var/lib/postgresql/PGVERSION/main to
+	 * /etc/postgresql/PG_VERSION/main.
+	 */
+	pgdataRelativePath = strstr(pgdata, "postgresql");
+	join_path_components(defaultConfigurationPath, "/etc", pgdataRelativePath);
+
+	/* true if we are working on debian/ubuntu installation */
+	configurationPathPresent = directory_exists(defaultConfigurationPath);
+	if (!configurationPathPresent)
+	{
+		log_error("Cannot find configuration file directory \"%s\"", defaultConfigurationPath);
+		return false;
+	}
+
+	join_path_components(defaultPosgresqlConfPath, defaultConfigurationPath, "postgresql.conf");
+	if (!move_file(defaultPosgresqlConfPath, posgresqlConfPath))
+	{
+		/* errors are already logged */
+		return false;
+	}
+
+	/* edit postgresql.conf to disable settings */
+	disable_configuration_parameters(posgresqlConfPath);
+
+	join_path_components(hbaConfPath, pgdata, "pg_hba.conf");
+	if(!file_exists(hbaConfPath))
+	{
+
+		log_trace("hba file does not exist : %s, need to copy", hbaConfPath);
+		join_path_components(defaultHbaConfPath, defaultConfigurationPath, "pg_hba.conf");
+
+		move_file(defaultHbaConfPath, hbaConfPath);
+	}
+
+	join_path_components(identConfPath, pgdata, "pg_ident.conf");
+	if(!file_exists(identConfPath))
+	{
+
+		log_trace("pg_ident.conf file does not exist : %s, need to copy", identConfPath);
+		join_path_components(defaultIdentConfPath, defaultConfigurationPath, "pg_ident.conf");
+
+		move_file(defaultIdentConfPath, identConfPath);
+	}
+
+	/* disable auto start in default configuration */
+	disableAutoStart(defaultConfigurationPath);
+
+	return true;
+}
+
+
+/*
+ * disable_configuration_parameters read postgresql.conf file and
+ * disables predefined configuration parameters inside.
+ */
+static bool
+disable_configuration_parameters(char *postgresqlConfPath)
+{
+	FILE *fileStream = NULL;
+	PQExpBuffer newConfContents = NULL;
+	char lineBuffer[BUFSIZE];
+	/*
+	 * configuration parameters can appear in any order, and we
+	 * need to check for patterns for NAME = VALUE and NAME=VALUE
+	 */
+	char *targetVariables[] = {
+			"data_directory ", "data_directory =",
+			"hba_file ", "hba_file=",
+			"ident_file ", "ident_file=",
+			"include_dir ", "include_dir=",
+			NULL};
```

Comment:
Why is `data_directory =` spelled with a space before the = sign where the other parameters have no space before the equal sign? Can we use a regex matching instead? We already have a `regexp_first_match` function in `parsing.c` after all.

A regex as the following should work?

    ^[ ]*(data_directory|hba_file|ident_file|include_dir)[ =]